src/librustc_save_analysis/lib.rs

```diff
@@ -620,45 +620,49 @@ impl<'l, 'tcx> SaveContext<'l, 'tcx> {
     }
 
     pub fn get_path_res(&self, id: NodeId) -> Res {
-        let hir_id = self.tcx.hir().node_id_to_hir_id(id);
-        match self.tcx.hir().get(hir_id) {
-            Node::TraitRef(tr) => tr.path.res,
-
-            Node::Item(&hir::Item { kind: hir::ItemKind::Use(path, _), .. }) => path.res,
-            Node::Visibility(&Spanned {
-                node: hir::VisibilityKind::Restricted { ref path, .. },
-                ..
-            }) => path.res,
-
-            Node::PathSegment(seg) => match seg.res {
-                Some(res) if res != Res::Err => res,
-                _ => {
-                    let parent_node = self.tcx.hir().get_parent_node(hir_id);
-                    self.get_path_res(self.tcx.hir().hir_id_to_node_id(parent_node))
-                }
-            },
+        if let Some(hir_id) = self.tcx.hir().opt_node_id_to_hir_id(id) {
```

Comment:
The goal is to resolve #71104 so that we can ultimately remove all uses of `opt_node_id_to_hir_id`. Can you add a `FIXME(#71004)` comment to this effect?

Reply:
Ahhh okay I wasn't aware of #71104. Yeah I'll add a comment in, that issue actually looks like something I'd really like to help on.

Reply:
In this specific case the `HirId` isn't present because the type is `tcx.types.err`. This must obviously be a more pervasive issue?

Reply:
I don't know a lot about #50928, but I would say yes.

Also, I normally hate style nitpicks on PRs, but I hate rightward drift even more. Can you return `Res::Err` early when `opt_node_id_to_hir_id` returns `None`?

```suggestion
        let hir_id = match self.tcx.hir().opt_node_id_to_hir_id(id) {
            Some(id) => id,
            None => return Res::Err,
        };
```

Reply:
Yeah hahaha that's an easy fix

Reply:
Should be sorted!